pallets/pallet-bonded-coins/src/tests/transactions/mint_into.rs

```diff
@@ -1 +1,705 @@
+use core::u128;
 
+use frame_support::{
+	assert_err, assert_ok,
+	traits::{
+		fungibles::Inspect,
+		tokens::{Fortitude, Preservation},
+	},
+};
+use frame_system::{pallet_prelude::OriginFor, RawOrigin};
+use sp_runtime::{assert_eq_error_rate, ArithmeticError, Permill, TokenError};
+
+use crate::{
+	curves::{polynomial::PolynomialParameters, Curve},
+	mock::{runtime::*, *},
+	types::{Locks, PoolStatus},
+	Error,
+};
+
+const MAX_ERROR: Permill = Permill::from_perthousand(1);
+
+#[test]
+fn mint_first_coin() {
+	let pool_id = calculate_pool_id(&[DEFAULT_BONDED_CURRENCY_ID]);
+
+	let curve = get_linear_bonding_curve();
+
+	let initial_collateral = 100u128;
+	let amount_to_mint = 1u128;
+	let expected_price = collateral_at_supply(amount_to_mint);
+
+	ExtBuilder::default()
+		.with_native_balances(vec![(ACCOUNT_00, ONE_HUNDRED_KILT)])
+		.with_collaterals(vec![DEFAULT_COLLATERAL_CURRENCY_ID])
+		.with_bonded_balance(vec![(DEFAULT_COLLATERAL_CURRENCY_ID, ACCOUNT_00, initial_collateral)])
+		.with_pools(vec![(
+			pool_id.clone(),
+			generate_pool_details(
+				vec![DEFAULT_BONDED_CURRENCY_ID],
+				curve,
+				true,
+				None,
+				None,
+				Some(DEFAULT_COLLATERAL_CURRENCY_ID),
+				None,
+			),
+		)])
+		.build()
+		.execute_with(|| {
+			let origin = RawOrigin::Signed(ACCOUNT_00).into();
+
+			assert_ok!(BondingPallet::mint_into(
+				origin,
+				pool_id.clone(),
+				0,
+				ACCOUNT_00,
+				amount_to_mint,
+				expected_price,
+				1
+			));
+
+			assert_eq_error_rate!(
+				<Test as crate::Config>::CollateralCurrencies::total_balance(
```

Comment:
Yes but I chose to interact with it everywhere via the fungibles traits to maintain loose coupling - it doesn't really matter but it's also how the pallet interacts with it.

Reply:
Yeah, it is the same. It would just reduce the lines and the complexity in the tests. 
I would prefer the change, but I will not stay in the way if you insist. Feel free to resolve the conversation.